payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/autocomplete/PlacesClientProxy.kt

```diff
@@ -0,0 +1,182 @@
+package com.stripe.android.ui.core.elements.autocomplete
+
+import android.content.Context
+import android.graphics.Typeface
+import android.text.style.StyleSpan
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.AutocompleteSessionToken
+import com.google.android.libraries.places.api.model.TypeFilter
+import com.google.android.libraries.places.api.net.FetchPlaceRequest
+import com.google.android.libraries.places.api.net.FindAutocompletePredictionsRequest
+import com.google.android.libraries.places.api.net.PlacesClient
+import com.stripe.android.BuildConfig
+import com.stripe.android.ui.core.R
+import com.stripe.android.ui.core.elements.autocomplete.model.AddressComponent
+import com.stripe.android.ui.core.elements.autocomplete.model.AutocompletePrediction
+import com.stripe.android.ui.core.elements.autocomplete.model.FetchPlaceResponse
+import com.stripe.android.ui.core.elements.autocomplete.model.FindAutocompletePredictionsResponse
+import com.stripe.android.ui.core.elements.autocomplete.model.Place
+import kotlinx.coroutines.tasks.await
+
+internal interface PlacesClientProxy {
+    suspend fun findAutocompletePredictions(
+        query: String?,
+        country: String,
+        limit: Int
+    ) : Result<FindAutocompletePredictionsResponse>
+
+    suspend fun fetchPlace(
+        placeId: String
+    ) : Result<FetchPlaceResponse>
+
+    companion object {
+        fun create(
+            context: Context,
+            googlePlacesApiKey: String,
+            isPlacesAvailable: IsPlacesAvailable = DefaultIsPlacesAvailable(),
+            clientFactory: (Context) -> PlacesClient = { Places.createClient(context) },
+            initializer: () -> Unit = { Places.initialize(context, googlePlacesApiKey) }
+        ): PlacesClientProxy {
+            return if (isPlacesAvailable()) {
+                initializer()
+                DefaultPlacesClientProxy(
+                    clientFactory(context)
+                )
+            } else {
+                UnsupportedPlacesClientProxy()
+            }
+        }
+
+        fun getPlacesPoweredByGoogleDrawable(
+            isSystemDarkTheme: Boolean,
+            isPlacesAvailable: IsPlacesAvailable = DefaultIsPlacesAvailable(),
+        ): Int? {
+            return if (isPlacesAvailable()) {
+                if (isSystemDarkTheme) {
+                    R.drawable.places_powered_by_google_dark
+                } else {
+                    R.drawable.places_powered_by_google_light
+                }
+            } else {
+                if (BuildConfig.DEBUG) {
```

Comment:
What if I'm on a debug build and I don't plan on using places? Will this crash my app?

Reply:
This won't crash your app if you don't supply a google places api key. We can remove this if we don't like the UX.